pkg/apis/autoscaling/v1alpha1/pa_defaults.go

```diff
@@ -30,19 +40,43 @@ func (r *PodAutoscaler) SetDefaults() {
 		// Default class to KPA.
 		r.Annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
 	}
-	// Default metric per class
+	// Default metric per class.
 	switch r.Class() {
 	case autoscaling.KPA:
 		if _, ok := r.Annotations[autoscaling.MetricAnnotationKey]; !ok {
 			r.Annotations[autoscaling.MetricAnnotationKey] = autoscaling.Concurrency
 		}
+		// KPA specific defaults.
+		if AutoscalerConfig != nil {
+			current := AutoscalerConfig.Current()
+			window := autoscaling.WindowAnnotationKey
+			target := autoscaling.TargetAnnotationKey
+			wpanic := autoscaling.WindowPanicPercentageAnnotationKey
+			tpanic := autoscaling.TargetPanicPercentageAnnotationKey
+			if _, ok := r.Annotations[window]; !ok && AutoscalerConfig != nil {
+				r.Annotations[window] = current.StableWindow.String()
+			}
+			if _, ok := r.Annotations[target]; !ok && AutoscalerConfig != nil {
+				r.Annotations[target] = asString(current.ContainerConcurrencyTargetDefault)
+			}
+			if _, ok := r.Annotations[wpanic]; !ok && AutoscalerConfig != nil {
+				r.Annotations[wpanic] = asString(current.WindowPanicPercentage)
+			}
+			if _, ok := r.Annotations[tpanic]; !ok && AutoscalerConfig != nil {
+				r.Annotations[tpanic] = asString(current.TargetPanicPercentage)
+			}
+		}
 	case autoscaling.HPA:
 		if _, ok := r.Annotations[autoscaling.MetricAnnotationKey]; !ok {
 			r.Annotations[autoscaling.MetricAnnotationKey] = autoscaling.CPU
 		}
 	}
 }
 
+func asString(v float64) string {
+	return strconv.FormatFloat(v, 'f', -1, 64)
```

Comment:
I am usually fan of strconv, but`fmt.Sprintf("%f", v)` -- is easier to much easier to read.

Reply:
Fair enough.  I'll change it.

Reply:
I decided to keep this using FormatFloat so that it doesn't add unnecessary zeros at the end.  It just look nicer in the test artifacts.